tests/test_microgrid/test_timeout.py

```diff
@@ -0,0 +1,185 @@
+"""Benchmark for microgrid data.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+import time
+from typing import Any, Iterator
+
+import grpc
+import pytest
+from frequenz.api.microgrid.microgrid_pb2 import (
+    ComponentData,
+    ComponentFilter,
+    ComponentIdParam,
+    ComponentList,
+    ConnectionFilter,
+    ConnectionList,
+    PowerLevelParam,
+    SetBoundsParam,
+)
+from google.protobuf.empty_pb2 import Empty  # pylint: disable=no-name-in-module
+from pytest_mock import MockerFixture
+
+from frequenz.sdk.microgrid.client import MicrogridGrpcClient
+
+from .mock_api import MockGrpcServer, MockMicrogridServicer
+
+# Timeout applied to all gRPC calls under test. It is expected after that the gRPC
+# calls will raise an AioRpcError with status code equal to DEADLINE_EXCEEDED.
+GRPC_CALL_TIMEOUT: float = 0.1
+
+# How much late a response to a gRPC call should be. It is used to trigger a timeout
+# error and needs to be greater than `GRPC_CALL_TIMEOUT`.
+GRPC_SERVER_DELAY: float = 0.3
+
+# How long a mocked Microgrid server should be running for a single test function,
+# before it gets shut down so that the tests can finish.
+GRPC_SERVER_SHUTDOWN_DELAY: float = 1.0
```

Comment:
Since we are awaiting for the function we are testing, I wonder if we really need to wait even more afterwards to shutdown the server? Can we just shut down immediately?

Reply:
I think the application would terminate immediately without the shut down, but there could be some asyncio warnings that there are pending tasks. So this is more of a graceful shutdown instead of exiting right away. I actually adopted it from other tests:
https://github.com/frequenz-floss/frequenz-sdk-python/blob/v0.x.x/tests/test_microgrid/test_client.py#L130